src/PlotlyPanel.tsx

```diff
@@ -108,19 +105,6 @@ export const PlotlyPanel: React.FC<Props> = (props) => {
     }
   }
 
-  if (
-    axisLabels.xAxis !== props.options.xAxis.field ||
-    !isEqual(axisLabels.yAxis, props.options.yAxis.fields) ||
-    !isEqual(axisLabels.yAxis2, props.options.yAxis2?.fields)
-  ) {
-    props.onOptionsChange({
-      ...props.options,
-      xAxis: { ...props.options.xAxis, field: axisLabels.xAxis },
-      yAxis: { ...props.options.yAxis, fields: axisLabels.yAxis },
-      yAxis2: { ...props.options.yAxis2, fields: axisLabels.yAxis2 },
-    });
-  }
-
```

Comment:
This was removed as we're no longer mutating the panel's options from within the plugin. Whatever fields the user selects will stay, even if they're nonexistent.